firebase_admin/_auth_utils.py

```diff
@@ -43,30 +43,29 @@ class PageIterator:
     def __init__(self, current_page):
         if not current_page:
             raise ValueError('Current page must not be None.')
+
         self._current_page = current_page
-        self._index = 0
+        self._iter = iter(self.items)
```

Comment:
Should we defer this until `next()` is called?

```py
def __init__(self):
  self._iter = None

def __next__(self):
  if self._iter is None:
    self._iter = iter(self.items)
```

Reply:
I've pushed a change on this patch that includes your suggestion. Thanks for taking time to review the patch and for your advice!